routes/rooms.go

```diff
@@ -127,12 +127,56 @@ func CreateEmergencyRoom(ctx *gin.Context) {
 // @Produce    json
 // @Param      authorization                   header      string                true    "Bearer: <TOKEN>"
 // @Param      id                              path        string                true    "Emergency Room's ID"
-// @Success    200                             {object}    GetSingleERResponse
-// @Failure    501                             {object}    HTTPErrorResponse
+// @Param      emergency-room                  body        PutERRequest          true    "ER to update"
+// @Success    200
+// @Failure    400                             {object}    HTTPErrorResponse
 // @Router     /er/{id}                        [patch]
 func UpdateEmergencyRoom(ctx *gin.Context) {
-	_ = ctx.Param("id")
-	SendError(ctx, http.StatusNotImplemented, errors.New("this endpoint is not implemented yet"))
+	log, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	erIdRaw := ctx.Param("id")
+	log.Debug().Str("update_id", erIdRaw)
+	erID, err := uuid.Parse(erIdRaw)
+	if err != nil {
+		SendError(ctx, http.StatusBadRequest, errors.New("invalid uuid"))
+		return
+	}
+
+	er := models.EmergencyRoom{
+		ID: erID,
+	}
+
+	// validate update body
+	body := PutERRequest{}
+	if err := ctx.ShouldBindJSON(&body); err != nil {
+		log.Warn().Err(err).Msg("validation failed")
+		SendError(ctx, http.StatusBadRequest, err)
+		return
+	}
+	log.Debug().Str("body", util.Formatted(body)).Send()
+
+	// convert department UUIDs into Departments
+	deps := make([]models.Department, len(body.Departments))
+	for i := range body.Departments {
```

Comment:
At this point we should write a function for this

Reply:
fixed